pkg/ccl/backupccl/restore_job.go

```diff
@@ -1716,19 +1716,14 @@ func remapPublicSchemas(
 
 // Resume is part of the jobs.Resumer interface.
 func (r *restoreResumer) Resume(ctx context.Context, execCtx interface{}) error {
+	p := execCtx.(sql.JobExecContext)
 	if err := r.doResume(ctx, execCtx); err != nil {
-		details := r.job.Details().(jobspb.RestoreDetails)
-		if details.DebugPauseOn == "error" {
-			const errorFmt = "job failed with error (%v) but is being paused due to the %s=%s setting"
-			log.Warningf(ctx, errorFmt, err, restoreOptDebugPauseOn, details.DebugPauseOn)
-
-			return jobs.MarkPauseRequestError(errors.Wrapf(err,
-				"pausing job due to the %s=%s setting",
-				restoreOptDebugPauseOn, details.DebugPauseOn))
+		log.Warningf(ctx, "The error was injected")
+		if err_inner := p.ExecCfg().JobRegistry.CheckPausepoint("restore.after_restore_failure"); err_inner != nil {
```

Comment:
go style nit: we generally don't use snake case; maybe `pauseErr` if you are trying to avoid shadowing?

(it would be shadowing, not overwriting, as long as you're using `:=` and only in the scope of the early return, but you might still want to avoid shadowing if we want to pass along the causing error too)